src/basho_bench_duration.erl

```diff
@@ -134,16 +135,20 @@ maybe_end(Return) ->
         infinity ->
             Return;
         Duration ->
-            case timer:now_diff(os:timestamp(), Start) of
-                Elapsed when Elapsed / 60000000 >= Duration ->
-                    ?CONSOLE("Stopping: ~p", [Elapsed]),
-                    {stop, normal, State};
-                Elapsed ->
-                    Timeout = round(Duration*60000 - Elapsed/1000),
-                    case tuple_size(Return) of
-                        2 -> {Reply, State, Timeout};
-                        3 -> {Reply, Message, State, Timeout}
-                    end
+            if Start /= undefined ->
```

Comment:
When did you find `Start` was undefined?

Reply:
When duration gen_server hasn't started, but worker gen_server has started already. Then it quickly finished due to empty keys in here: https://github.com/cloudant/basho_bench/blob/cloudant/src/basho_bench_duration.erl#L78.